File: typescript/rna-transcription/rna-transcription.ts
class Transcriptor {
  toRna( sequenceString: string ): string {

    return sequenceString.split('')
    .map(code => {
      switch(code){
        case('G'):
          return 'C';
        case('C'):
          return 'G';
        case('T'):
          return 'A';
        case('A'):
          return 'U';
        default:
          throw new Error('Invalid input DNA.');
      }
    })
    .join('');

  }
}

export default Transcriptor

File: python/rna-transcription/rna_transcription.py

def transcribe(character):
    if character == 'C':
        return 'G'
    elif character == 'G':
        return 'C'
    elif character == 'T':
        return 'A'
    elif character == 'A':
        return 'U'

def to_rna(dna_strand):
    
    return( ''.join(map(transcribe, dna_strand)) )

    
